### DISCUSSION TITLE
Gestione relais

### DISCUSSION THREAD

**Discussion by Jocman** [indomus/forum#79] [Hardware & Firmware]
Ciao a tutti.
Nuovo del forum (e anche di HomeAssistant)
Poichè sto iniziando a...smanettare HA, sto cercando di schematizzare ciò che vorrei pilotare domoticamente a casa mia.
Mi piacerebbe utilizzare le schede ESP32, le sto studiando un pò anche perchè vorrei sfruttare tutti i loro GPIO per pilotare più sensori/attuatori (situati nel medesimo ambiente) impegnando un solo IP.
Ma di questo magari chiederò in seguito.
Il mio dubbio attualmente riguarda il pilotaggio delle mie luci di casa (comincerò da quelle).
Stavo valutando (in attesa di riuscire con le ESP32) di utilizzare i prodotti Sonoff (che tra l'altro sono basati su ESP); ma documentandomi in rete sono giunto alla conclusione che in effetti i Sonoff (ad esempio il basic) potrebbero essere collegati solo all'interruttore di accensione ON/OFF.
Peccato che il 99% delle mie luci siano pilotate da relais, e se non ho capito male il Sonoff non è utilizzabile in questo caso, in quanto non ho un interruttore ma un pulsante momentaneo.
I relais (7) sono montati su barra DIN direttamente nel quadro elettrico.
E' giusta la mia perplessità o invece posso tranquillamente utilizzare i sonoff?
In alternativa che soluzione potrei adottare?

Andrea

**Comment by dante70** [comment]
ciao a tutti e buona Pasqua.
per quanto riguarda la gestione delle luci  il mio consiglio è quello di  di non utilizzare ne 'esp32 ne 2son off ma dei moduli shelly 2.5 .
Il motivo è presto detto :nascono per essere "pilotati" direttamente con la tensione di rete e cioe' quella che percorre sicuramente i tuoi pulsanti che al momento vanno ad eccitare i rele' modulari che hai su barra din.
Gli son off salvo se non erro l'ultimo uscito prevedono di essere comandati da una bassa tensione generata direttamente dal modulo.(si deve  chiudere/aprire il contatto S1-S2 ).Il problema è che (toccato con mano di persona) devi modificare l'impianto facendo girare nei vari pulsanti questa bassa tensione con il risultato che oltre alla mole di lavoro da fare c'è anche il problema  che se i cavi sono abbastanza lunghi possono essere soggetti ad interferenze con indesiderati interventi del relè anche se non abbiamo cambiamenti di stato dell'ingresso.Inizialmente ho costruito qualche modulo con esp32 e D1 mini ma per usi con sonde ecc ma quando ho provato per una tapparella motorizzata nasceva il problema sopradescritto.Ho usato quindi a quel punto degli shelly 2.5 che avendo 2 rele'  e 2 ingressi possono gestire 2 punti luci  indipendenti.Nel tuo caso per 7  luci te la caveresti con 4 apparecchietti.
Il costo degli shelly e' maggiore rispetto al sonoff mini ma è compensato dai vantaggi che ho indicato.Anche itead e' arrivata con il dual R3 ma io ancora non lo ho provato e non ho letto bene se è equivalente allo shelly2.5.

**Reply by Jocman** [reply]
Beh, a parte ringraziarti per il suggerimento, purtroppo mi hai dato una....notizia poco piacevole.
Non fosse altro che anni fa avevo comprato del moduli relays da 8, che poi non sono serviti (ma per altri motivi) e speravo di utilizzarli in questo modo....
Darò sicuramente un'occhiata agli shelly (anche io avevo una mezza idea di usare quelli, ma sai, 8 relays da pilotare sono tantini (e sono solo una parte dei relays luci che ho.....)
Comunque, penso concordo con te che il gioco varrebbe la candela
